aws-ssm-patchbaseline/src/main/java/software/amazon/ssm/patchbaseline/ListHandler.java

```diff
@@ -122,6 +123,10 @@ private DescribePatchBaselinesResponse describePatchBaselinesResponse(final Stri
                 .stream().map(baseline -> baseline.baselineId()).collect(Collectors.toList());
 
         for (String baselineId : baselineIdList) {
+            if(baselineId.contains(File.separator)){
```

Comment:
i still don't understand this. 
My argument is that:

baselines returned as an arn  arn:aws:ssm:us-east-2:733109147000:patchbaseline/pb-0e392de35e7c563b7 instead of only baseline id pb-0e392de35e7c563b7, means this baseline is owned by Patch Manager. 

and later it's passed to readHandler to read the content of the baseline, for Patch owned baseline, you need to input the whole arn arn:aws:ssm:us-east-2:733109147000:patchbaseline/pb-0e392de35e7c563b7 , instead of pb-0e392de35e7c563b7 to succeed.

Reply:
So with your current code, the baseline arn is truncated to baseline Id, when it's passed to readHandler, it would definitely fail, because the baselineId doesn't exist.

Reply:
so this means it would fail, i don't understand how did it succeed

Reply:
for example, the following is a patch owned baseline in us-east--1, 
 arn:aws:ssm:us-east-1:075727635805:patchbaseline/pb-03e3f588eec25344c 

**aws ssm get-patch-baseline --baseline-id pb-03e3f588eec25344c** would run into the following exception: 

An error occurred (DoesNotExistException) when calling the GetPatchBaseline operation: Patch Baseline to be retrieved does not exist.

**aws ssm get-patch-baseline --baseline-id arn:aws:ssm:us-east-1:075727635805:patchbaseline/pb-03e3f588eec25344c** is the request input

{
    "BaselineId": "arn:aws:ssm:us-east-1:075727635805:patchbaseline/pb-03e3f588eec25344c",
    "Name": "AWS-CentOSDefaultPatchBaseline",
    "OperatingSystem": "CENTOS",
    "GlobalFilters": {
        "PatchFilters": []
    },
    "ApprovalRules": {
        "PatchRules": [
            {
                "PatchFilterGroup": {
                    "PatchFilters": [
                        {
                            "Key": "PRODUCT",
                            "Values": [
                                "*"
                            ]
                        }
                    ]
                },
                "ComplianceLevel": "UNSPECIFIED",
                "ApproveAfterDays": 7,
                "EnableNonSecurity": true
            }
        ]
    },
    "ApprovedPatches": [],
    "ApprovedPatchesComplianceLevel": "UNSPECIFIED",
    "ApprovedPatchesEnableNonSecurity": false,
    "RejectedPatches": [],
    "RejectedPatchesAction": "ALLOW_AS_DEPENDENCY",
    "PatchGroups": [],
    "CreatedDate": 1522345092.515,
    "ModifiedDate": 1522345092.515,
    "Description": "Default Patch Baseline for CentOS Provided by AWS.",
    "Sources": []
}

Reply:
now i think about it, adding a readHandler inside ListHandler may not be as appropriate, we can return the value of BaselineIdentities returned from describe-patch-baselines directly. 
Can you try this out?

Reply:
yes we have tired the same.
But if we tried to return "describe-patch-baselines " it doesnot contains the pactchbaseline full details i.e: what you have mentioned above and It will contain the data as below.

DescribePatchBaselinesResponse(BaselineIdentities=[
  PatchBaselineIdentity(BaselineId=arn: aws: ssm: us-east-1: 075727635805: patchbaseline/pb-03e3f588eec25344c,
  BaselineName=AWS-CentOSDefaultPatchBaseline,
  OperatingSystem=CENTOS,
  BaselineDescription=DefaultPatchBaselineforCentOSProvidedbyAWS.,
  DefaultBaseline=true)
])

Since, we are taking the baseline id from describe patch baseline response and passing the baseline id to  getpatchbasline functionality to get the baseline details as you mentioned.
Now the Listhandler process is working as you mentioned.